src/main/java/core/basesyntax/WorkWithFile.java

```
@@ -1,7 +1,67 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+
 public class WorkWithFile {
+    public static final int INDEX_OF_CATEGORY = 0;
+    public static final int INDEX_OF_VALUE = 1;
+    public static final int LENGTH_OF_ARRAY = 2;
+    public static final int SUPPLY = 0;
+    public static final int BUY = 1;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        try {
+            int[] statistics = parseInformationFromFile(fromFileName);
+            String report = buildReport(statistics);
+            writeReport(toFileName, report);
+        } catch (IOException e) {
+            throw new RuntimeException("I/O error while processing files: "
+                    + fromFileName + " or " + toFileName, e);
```

getStatistic wraps both read and write operations in a single try/catch and throws a generic message: "I/O error while processing files: ... or ...". The checklist recommends separating operations and using precise messages ("Don't open nested connections to files" and "Don't ignore exceptions"). Read and aggregate in one try-with-resources; then write in a separate try/catch with a message like `"Can't write report to file: " + toFileName` and include the caught exception. (Checklist: "Don't open nested connections to files" and "Don't ignore exceptions") .